### PR TITLE
protocol: respect requests from partition

### PR DESCRIPTION
**Pull request by deepak1556**
Protocol module currently doesnt respect partitions. This just adds a `fromPartition` method similar to `session`, internally `protocol` is now wired as a property of `session`.

### CODE REVIEW

**Comment by deepak1556** [comment]
@zcbenz any clue about why this crash happens https://travis-ci.org/atom/electron/jobs/110389798 , just made a temp fix. Would like resolve the original cause.

**Comment by zcbenz** [comment]
I don't think we need the `protocol.fromPartition` API, we can just explicitly document `session.protocol` and people will know how to use it for multi-session.

> @zcbenz any clue about why this crash happens https://travis-ci.org/atom/electron/jobs/110389798 , just made a temp fix. Would like resolve the original cause.

I think it is because calling `app.exit` crashed the main process.

**Comment by deepak1556** [comment]
Fixed, thanks!

File: docs/api/session.md [hunk 1]
```diff
@@ -523,3 +523,24 @@ The `listener` will be called with `listener(details)` when an error occurs.
...
+
+#### `ses.protocol`
+
+Returns an instance of [protocol](protocol.md) module for this session.
+
+```javascript
+const electron = require('electron');
+const app = electron.app;
+const path = require('path');
+const protocol = session.fromPartition(partitionName).protocol;
```
Comment (zcbenz) [inline docs/api/session.md]: Creating session before the ready event is not recommended, for now it is causing crash on exit.

Comment (deepak1556) [inline docs/api/session.md]: Fixed.

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**